src/css/content/icons.css

```diff
@@ -132,7 +132,11 @@ a:not([role="button"]):has(use[href*="icon"]) {
   --ds-icon-color: var(--link);
 }
 
-:is(button, input[type="submit"], input[type="button"], input[type="reset"], a[role="button"]):hover,
-:is(button, input[type="submit"], input[type="button"], input[type="reset"], a[role="button"]):active {
+:is(button, input[type="submit"], input[type="button"], input[type="reset"], a[role="button"]).warning {
+  --ds-icon-color: var(--color-inverse);
+}
+
+:is(button, input[type="submit"], input[type="button"], input[type="reset"], a[role="button"]):not(.warning):hover,
+:is(button, input[type="submit"], input[type="button"], input[type="reset"], a[role="button"]):not(.warning):active {
```

Comment:
Ville gerne slippe for at have `:not(.warning)` i de generelle rules. 
Foretrækker at undtagelser fra de generelle styles er samlet for sig. Så noget á la:
```
:is(button, input[type="submit"], input[type="button"], input[type="reset"], 
a[role="button"]).warning {
  --ds-icon-color: var(--color-inverse);

  &:hover,
  &:active {
   --ds-icon-color: var(--color-inverse);
 }
}
```
Måske har du forsøgt det og ikke fundet det anvendeligt?